### ISSUE TITLE
Adding the Mac mosaic blur to sidebars

### ISSUE THREAD

**Issue by bluejamesbond** [radiant-player/radiant-player-mac#613]
I have added the mosaic blur to the app to mimic the Apple style. It looks like this:

**Left Sidebar**
![image](https://cloud.githubusercontent.com/assets/376453/18812547/36eb51b2-828d-11e6-9e33-7f1e4e6355b0.png)

**Header and Right Sidebar**
![image](https://cloud.githubusercontent.com/assets/376453/18812556/7ebe0db8-828d-11e6-9232-a9ed1f524fe2.png)

I have made changes in my fork. I will be happy to test and then make a PR if this is something you are interested in.

**Comment by BarakaAka1Only** [comment]
I like that. 

I took a look at your changes on your fork i see you applied:

`-webkit-backdrop-filter`

Since its only in 3 areas,  i think it would be best if it were in a preference setting so that you can choose with or without the blur then from there, if any other sections needs blurs etc it can be added in one place than going through the main file where its all crowded.

Either way nice 👍

**Comment by bluejamesbond** [comment]
@BarakaAka1Only Awesome, I can try to do that. Any idea on how to read preference settings via css? Additionally, if you notice the window (of the first picture), you can see a light white border. Has there been any effort to hide that?

PS. I updated the images to give you a better idea of the header

**Comment by BarakaAka1Only** [comment]
That is interesting, looks like something with #main 

try something with border-top on that id as that is the main window

Also there isn't a way to read the preference in css because the file, the css is added to the dom on load thats where you would need to add a your own set take a look at the Styles folder [DarkCyanStyle.h/DarkCyanStyle.m/ApplicationStyle.m] 

Last thing the `-webkit-backdrop-filter` does not exist on Yosemite and below its 
`-webkit-filter:`

;)

**Comment by MarshallOfSound** [comment]
@BarakaAka1Only `backdrop-filter` and `filter` are two very different properties.

The first applies to the area behind a DOM Node.  The other applies to the DOM Node itself

**Comment by BarakaAka1Only** [comment]
Whoops a mistype just, didn't notice i meant to say `backdrop-filter` my bad

**Comment by bluejamesbond** [comment]
@BarakaAka1Only Yes. I was aware of the compatibility issue but it has a very safe fallback which is basically no blur. As a result, the "background blur" preference will only be visible to who are able to run this feature i.e. > Yosemite. What do you think?

**Comment by BarakaAka1Only** [comment]
Sounds reasonable because of the fact when doing a version test comes out as version 5 when it only works 9 and above 

`
version: "5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/600.8.9 (KHTML, like Gecko)"
`

Not sure if there is a way to change the framework version to use 9 or something but i don't think it's possible.

So i think that would be best for now 👍

**Comment by bluejamesbond** [comment]
How about?

```js
"webkitBackdropFilter" in document.body.style // true
```

**Comment by BarakaAka1Only** [comment]
That comes out false because the version that Yosemite is using is 5 when backdrop is only supported 9 and above 

hence `@supports (-webkit-backdrop-filter: none) or (backdrop-filter: none) {}`
and 

    var vendorPrefixes = ",Webkit,Moz,O,ms,Khtml".split(",");
    
    function supports(feature) {
    		var support = false;
    		for (var i = 0; i < vendorPrefixes.length && !support; i++) {
    				support = vendorPrefixes[i] + feature in document.body.style
    		}
    		return support;
    }
    
    supports("webkitBackdropFilter"); // false
    supports("BackdropFilter");  // false

_credits [codepen](http://codepen.io/netsi1964/pen/zGVwdV) for the js_

**Comment by bluejamesbond** [comment]
Added a preference in the appearance option. 

![image](https://cloud.githubusercontent.com/assets/376453/18813198/6d8d65e8-82a6-11e6-9a52-786897ea7ed9.png)

**Comment by BarakaAka1Only** [comment]
Tested on Sierra and works as expected 👍  only thing since the css update, DarkCyanStyle now get sets after spotify-black as it inherits the styles from there and adds the cyan color(s) so as of right now with the blur enabled the DarkCyanStyle on the `#material-app-bar` is black other than that awesome 👍

**Comment by bluejamesbond** [comment]
@BarakaAka1Only Thanks. There are issues still to resolve. These issues will be resolved over the next couple of days after some testing. I'll will update here as soon as it seems stable.

**Comment by BarakaAka1Only** [comment]
Looking forward to it.

**Comment by BarakaAka1Only** [comment]
To add on if OS is Yosemite and below,

i played around with the AppearancePreferencesViewController files and pretty much if the case is true via `is Yosemite and below` it will hide and rearrange the preferences settings in that view

so it will look like 

![image](https://cloud.githubusercontent.com/assets/2947494/18813763/3f87e770-82cb-11e6-90b3-99cad427986e.png)
  
while if  El Capitan and above will have the background blur option ;)  


AppearancePreferencesViewController.h added:


    @property (assign) IBOutlet NSButton *BlurPrefs;
    @property (assign) IBOutlet NSButton *AppearancePrefs;
    @property (assign) IBOutlet NSButton *StyleChoosePrefs;
    @property (assign) IBOutlet NSTextField *StyleChooseTextPrefs;

AppearancePreferencesViewController.m added:

    @synthesize BlurPrefs;
    @synthesize AppearancePrefs;
    @synthesize StyleChoosePrefs;
    @synthesize StyleChooseTextPrefs;
    
    - (void)awakeFromNib
    {
        self.sortDescriptors = @[ [NSSortDescriptor sortDescriptorWithKey:@"name" ascending:YES] ];
    
        if (floor(NSAppKitVersionNumber) <= NSAppKitVersionNumber10_9) {
            [BlurPrefs setHidden:TRUE];
            [AppearancePrefs setFrameOrigin:NSMakePoint(AppearancePrefs.frame.origin.x, AppearancePrefs.frame.origin.y + 15)];
            [StyleChoosePrefs setFrameOrigin:NSMakePoint(StyleChoosePrefs.frame.origin.x, StyleChoosePrefs.frame.origin.y + 5)];
            [StyleChooseTextPrefs setFrameOrigin:NSMakePoint(StyleChooseTextPrefs.frame.origin.x, StyleChooseTextPrefs.frame.origin.y + 5)];
        } else if (floor(NSAppKitVersionNumber) <= NSAppKitVersionNumber10_10_Max) {
            [BlurPrefs setHidden:TRUE];
            [AppearancePrefs setFrameOrigin:NSMakePoint(AppearancePrefs.frame.origin.x, AppearancePrefs.frame.origin.y + 15)];
            [StyleChoosePrefs setFrameOrigin:NSMakePoint(StyleChoosePrefs.frame.origin.x, StyleChoosePrefs.frame.origin.y + 5)];
            [StyleChooseTextPrefs setFrameOrigin:NSMakePoint(StyleChooseTextPrefs.frame.origin.x, StyleChooseTextPrefs.frame.origin.y + 5)];
        }
        
    }

Then added the Referencing outlets in PreferencesWindowController.xib

Now the condition above is long yes, i tried few ways to shorten it to one but it wasn't applying as it kept being true on all OS versions but as of now this works and i've tested it on all OS's ranging from
Mavericks to Sierra 

Doing this addition will give less opened issues with users saying this option doesn't work etc if on Yosemite and below.

_sorry for the redundancy_

**Comment by bluejamesbond** [comment]
@BarakaAka1Only Ok Ill put this code in. By any chance do you know how i can bundle this application (with the new blur) so i can share with some friends who can then test it out on their machines?

**Comment by BarakaAka1Only** [comment]
Looks amazing!  And yeah, when you build the project it automatically compiles it to a runnable application you would have to go to the build folder here `~/Library/Developer/Xcode/DerivedData`

from there you will see radiant-player-mac- with a bunch of random numbers / letter at the end something like `radiant-player-mac-dshnwexnfcmktxegznoofisfgxhb`  go into the folder then 
Build -> Products -> Debug and you will see Radiant Player.app you can compress it to a zip and send it off.

**Comment by jacobwgillespie** [comment]
Hey, what's the status of this?  Looks awesome, I would love to get this merged.

**Comment by bluejamesbond** [comment]
There are some performance issues to take into consideration. Additionally, there are some CSS changes needed to ensure that it works well across all the pages. But other than that it pretty much there - just got busy at work.
